sched/task/task_posixspawn.c

```diff
@@ -410,7 +410,7 @@ int posix_spawn(FAR pid_t *pid, FAR const char *path,
    * for use within the OS.
    */
 
-  ret = nx_waitpid(proxy, &status, 0);
+  ret = nx_waitpid((pid_t)proxy, &status, 0);
```

Comment:
How about keeping `proxy` as `pid_t` and instead receive the return from `kthread_create` with the `ret` local?

Reply:
Yes. This can be an option. I will implement the change